src/particles.cpp

```diff
@@ -679,6 +755,16 @@ void ParticleManager::addNodeParticle(IGameDef* gamedef,
 	addParticle(toadd);
 }
 
+u32 ParticleManager::getSpawnerId()
+{
+	for (u32 id = 0;; ++id) { // look for unused particlespawner id
```

Comment:
infinite loop possible in a very very rare case, please handle it

also this function should be const